src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -382,6 +383,15 @@ fn check_terminator(
                         .into(),
                     ));
                 }
+                if tcx.fn_sig(def_id).abi() == RustIntrinsic
+                    && tcx.item_name(def_id) == sym::transmute
+                    && !feature_allowed(tcx, def_id, sym::transmute_in_const_fn)
```

Comment:
This is the fun part. It works, but it's absolutely not scalable to more functions or intrinsics.

Reply:
Maybe add a comment `// HACK` or so, just to document that point.